### PR TITLE
Fixes #5806 Implemented "Refresh" button to clear the cache and reload the Nearby map

### LINKED ISSUE #5806: Nearby: "Refresh" button to clear the cache and reload the Nearby map
It would be good to have a "Refresh" icon button in the title bar of Nearby.
Tapping on it would:

1. Empty the Nearby local cache. Context: Pins are saved in a cache so that Nearby loads faster. Even if another user adds a picture for that pin, the pin does not change color.
2. Reload the Nearby map. It would not strictly necessary if all things were perfect, but Nearby is a fast-moving feature which often (always?) has some bugs, so I very often find myself restarting the app just to reload this map (for instance because the blue dot does not follow me).

### PR DESCRIPTION
**Pull request by noahvendrig**
The nearby map can be buggy, and a reload feature is much more efficient than reloading the whole app.
This pull request solves this by adding a refresh button which will empty the nearby cache, and then reload the map.

Fixes #5806

**What changes did you make and why?**
1. After cloning the app would not run in my environment, and I have noticed that this is a common issue with other new maintainers. To solve this I had to: Removed suspend from affected DAO files and funcs, and changed to (Kotlin v1.9.22) and (Kotlin compiler v1.5.8)

2. `NearbyParentFragment.java`
    This is where the clicking of the refresh button is handled, which calls `emptyCache()`, which will then call `reloadMap()` to complete the refresh. Main functionality has been implemented here.

4. `nearby_fragment_menu.xml`, `ic_refresh_24dp_nearby.xml`
    Created new refresh icon and added it to the nearby menu, so that it is displayed in the top header.

5. `NearbyParentFragment.java`, `PlacesRepository.java`, `PlacesLocalDataSource.java`, `PlaceDao.java`
    These files were changed to implement emptying the nearby cache

6. `NearbyParentFragment.java`
    Functionality for reloading the map is inside `reloadMap()`

**Tests performed (required)**

1. Tested  'app' profile on Huawei Pura 70 with API level 32
3. Tested  'app' profile on Virtual Device "Medium Phone" with API level 34

**Screenshots (for UI changes only)**
[Screen_recording_20241025_152616.webm](https://github.com/user-attachments/assets/dac0029c-ee91-4af8-baf2-9af21958b1ba)

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java [hunk 1]
```diff
@@ -1158,6 +1171,48 @@ public void populatePlaces(final LatLng currentLatLng,
...
 
+    /**
+     *  Reloads the Nearby map
+     *  Clears all location markers, refreshes them, reinserts them into the map.
+     *
+     */
+    private void reloadMap() {
+        clearAllMarkers(); // Clear the list of markers
+        binding.map.getController().setZoom(ZOOM_LEVEL); // Reset the zoom level
+        binding.map.getController().setCenter(lastMapFocus); // Recenter the focus
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:1182]: Refreshing seems to move the map to the place I am in physically, is it expected?

https://github.com/user-attachments/assets/76cd2cad-4ce3-4048-91d8-83c048e3eb18

**Review by nicolas-raoul — APPROVED** [review]
Works great, in both day and night mode.

File: app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java (hunk 1 above)
Comment (noahvendrig) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:1182]: > Refreshing seems to move the map to the place I am in physically, is it expected?
> 
>  screen-20241025-141156.mp4

My colleague @MarcusBarta worked on the map reload part of this feature, and he has told me that this is expected.

Comment (nicolas-raoul) [reply app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java:1182]: Understood! I created https://github.com/commons-app/apps-android-commons/issues/5906 as an enhancement for this.

**Pull request merged**